specification/profiling.md

```diff
@@ -0,0 +1,169 @@
+        - [LogRecord Fields](#logrecord-fields)
+            * [Call Stack Format](#call-stack-format)
+- [References](#references)
+
+## Activation
+
+Agents may choose to activate or deactivate their profiler by default.
+A configuration setting MUST be provided to allow overriding the default: 
+
+* `SPLUNK_PROFILER_ENABLED` environment variable
+
+The default setting for `SPLUNK_PROFILER_ENABLED` MUST be clearly documented. 
+The agent SHOULD NOT allow changing the setting at runtime, and the initial 
+setting SHOULD be used for the entire lifespan of the application run.
+
+An agent whose profiling capability is deactivated MUST NOT introduce
+additional profiling-based overhead. It also MUST NOT ingest profiling-based
+data.
+
+## Call Stack Sampling
+
+An agent that has profiling capabilities MUST be able to sample call stacks
+at a fixed interval. This default interval MUST default to 1 second.
+
+If a language agent runtime supports threading, stacks MUST be sampled across
+all process threads. The samples for all threads SHOULD be taken instantaneously
+and, in the event that this is not feasible, MUST be taken as close together
+as possible. If the samples are taken consecutively, then the profiler MUST 
+use a consistent ordering strategy (such as thread name or ID) when sampling all
+threads.
+
+### Call Stack Filtering
+
+Various runtimes MAY contain internal and other threads that are undesirable
+to include for profiling. This could include threads that are internal to 
+runtime behavior or agent internal workings. The choice of which threads are 
+undesirable is implementation specific and not defined.
+
+The profiler SHOULD not collect call stacks from undesirable threads.
+If this is not possible, the profiler SHOULD filter these out afterward 
+and omit these call stacks from ingest. 
+
+### Call Stack Span Association
+
+When a call stack is sampled during the execution of a span scope,
+the profiler MUST be able to associate the call stack to the span.
+This association SHOULD happen as close to the sampling point as feasible,
+but MAY occur later in a processing pipeline.
+
+After the association has been made, the `TraceId` and `SpanId` fields of the 
+LogRecord MUST be populated (see below).
+
+### Call Stack Ingest
+
+Call stacks MUST be ingested as [OpenTelemetry Logs](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs).
+The logs containing profiling data MUST be sent via OTLP/gRPC. 
+
+By default, the agent MUST send logs to the value in the `OTEL_EXPORTER_OTLP_ENDPOINT` environment
+variable. If this variable is not set, then the agent must default back to the gRPC/OTLP
+default (https://localhost:4317).
+
+The agent must allow the destination for profiling logs to be overridden with 
+the environment variable SPLUNK_PROFILER_LOGS_ENDPOINT.
+
+Agents SHOULD reuse persistent gRPC/OTLP connections from other components (traces, metrics).
```

Comment:
Java doesn't do this yet, but I want it to...because I believe that gRPC channels are backed by a connection pool and thread pool and not reusing takes up extra resources.